### PR TITLE
discovery 404s

### PR DESCRIPTION
**Pull request by AaronLaBrie**
Discovery now 404's when a config is not found. Adjust the lib to support that.

---
This change is [Reviewable](https://reviewable.io/reviews/contextio/contextio-go/22)

### CODE REVIEW

File: ciolite/request.go [hunk 1]
```diff
@@ -25,7 +25,7 @@ type clientRequest struct {
 }
 
 // doFormRequest makes the actual request
-func (cio CioLite) doFormRequest(request clientRequest, result interface{}) error {
+func (cio CioLite) doFormRequest(request clientRequest, result interface{}) (error, int) {
```
Comment (AaronLaBrie) [inline ciolite/request.go]: Most of the changes are just to handle this function signature change

File: ciolite/lite_discovery.go [hunk 2]
```diff
@@ -41,7 +39,7 @@ type GetDiscoveryIMAPResponse struct {
 // GetDiscovery attempts to discover connection settings for a given email address.
 // queryValues requires Email to be set.
 // 	https://context.io/docs/lite/discovery#get
-func (cioLite CioLite) GetDiscovery(queryValues GetDiscoveryParams) (GetDiscoveryResponse, error) {
+func (cioLite CioLite) GetDiscovery(queryValues GetDiscoveryParams) (*GetDiscoveryResponse, error) {
```
Comment (AaronLaBrie) [inline ciolite/lite_discovery.go]: Return `nil, nil` if discovery 404's. I like this pattern generally, but it is a little awkward to have this be the only pointer response.

A possible solution would be to define an `ErrNotFound` and always return that from the request functions on 404s, but that's a larger change than I wanted to make right this moment.

File: ciolite/request.go (hunk 1 above)
Comment (dmlyons) [reply ciolite/request.go]: Traditionally the error is the last thing returned by a func or method, it feels kinda weird having it the first thing with another thing after it.

Comment (AaronLaBrie) [reply ciolite/request.go]: Quite right. Clearly more coffee is needed.

**Comment by dmlyons** [comment]
10:58 $ go test ./...
# github.com/ReturnPath/godzilla/services/cio-discovery
./main.go:75:53: syntax error: unexpected newline, expecting comma or )
./main.go:77:3: syntax error: unexpected ), expecting expression
✘-2 ~/go/src/github.com/ReturnPath/godzilla/services/cio-discovery dlyons [al-discovery|…5]
10:59 $ go build .
# github.com/ReturnPath/godzilla/services/cio-discovery
./main.go:75:53: syntax error: unexpected newline, expecting comma or )
./main.go:77:3: syntax error: unexpected ), expecting expression

**Pull request closed without merging**